zc_events/client.py

```diff
@@ -116,49 +118,54 @@ def wait_for_response(self, response_key):
         response = self.redis_client.blpop(response_key, 5)
         return response
 
-    def handle_request_event(self, event, viewset=None, relationship_viewset=None):
+    def handle_request_event(self, event, view=None, viewset=None, relationship_viewset=None):
         """
         Method to handle routing request event to appropriate view by constructing
         a request object based on the parameters of the event.
         """
-        request = create_django_request_object(event)
-
-        # Call the viewset passing the appropriate params
-        if event.get('id') and event.get('relationship'):
-            result = relationship_viewset.as_view()(request, pk=event.get('id'),
-                                                    related_field=event.get('relationship'))
-        elif request.method == 'GET' and event.get('id') and event.get('related_resource'):
-            result = viewset.as_view({'get': event.get('related_resource')})(request, pk=event.get('id'))
-        elif request.method == 'GET' and event.get('id'):
-            result = viewset.as_view({'get': 'retrieve'})(request, pk=event.get('id'))
-        elif request.method == 'PUT' and event.get('id'):
-            result = viewset.as_view({'put': 'update'})(request, pk=event.get('id'))
-        elif request.method == 'PATCH' and event.get('id'):
-            result = viewset.as_view({'patch': 'partial_update'})(request, pk=event.get('id'))
-        elif request.method == 'DELETE' and event.get('id'):
-            result = viewset.as_view({'delete': 'destroy'})(request, pk=event.get('id'))
-        elif request.method == 'GET':
-            result = viewset.as_view({'get': 'list'})(request)
-        elif request.method == 'POST':
-            result = viewset.as_view({'post': 'create'})(request)
-        elif request.method == 'OPTIONS' and event.get('id'):
-            result = viewset.as_view({
-                'get': 'retrieve',
-                'put': 'update',
-                'patch': 'partial_update',
-                'delete': 'destroy'
-            })(request, pk=event.get('id'))
-        elif request.method == 'OPTIONS':
-            result = viewset.as_view({
-                'get': 'list',
-                'post': 'create',
-            })(request)
+        kwargs_copy = copy.deepcopy(event)
+        request, kwargs = create_django_request_object(kwargs_copy)
```

Comment:
My suggestion is to pop the keys you want from for the request before calling the method. Pass them as args to `create_django_request_object`. This way you don't need to copy the `event` and it's more clear what object the `kwargs` is without knowing the internals of `create_django_request_object`. it will also have all key pops in one method instead of different places and the `create_django_request_object` will be just creating a request object.